ArchiSteamFarm/IPC/Integration/ApiAuthenticationMiddleware.cs

```diff
@@ -87,16 +87,32 @@ internal sealed class ApiAuthenticationMiddleware {
 				throw new InvalidOperationException(nameof(ClearFailedAuthorizationsTimer));
 			}
 
+			IPAddress? clientIP = context.Connection.RemoteIpAddress;
+
+			if (clientIP == null) {
+				throw new InvalidOperationException(nameof(clientIP));
+			}
+
 			string? ipcPassword = ASF.GlobalConfig != null ? ASF.GlobalConfig.IPCPassword : GlobalConfig.DefaultIPCPassword;
 
 			if (string.IsNullOrEmpty(ipcPassword)) {
-				return HttpStatusCode.OK;
-			}
+				if (IPAddress.IsLoopback(clientIP)) {
+					return HttpStatusCode.OK;
+				}
 
-			IPAddress? clientIP = context.Connection.RemoteIpAddress;
+				if (Startup.KnownNetworks.IsEmpty) {
```

Comment:
`KnownNetworks` doesn't get reloaded when changing `IPC.config` (despite config itself being reloaded), you can add `IOptions<ForwardedHeadersOptions>` argument to middleware constructor and retrieve `KnownNetworks` from there.

Reply:
I'm actually wrong here, it does get reloaded - however passing through constructor seems a little bit cleaner IMO.